### PR TITLE
feat: #344 Add logger that show amount of request operations

### PR DESCRIPTION
**Pull request by Ekaterina1994**

### CODE REVIEW

File: mw-webapp/src/logic/wayPage/reportsTable/DayReportsTable.tsx [hunk 1]
```diff
@@ -45,8 +45,7 @@ export const DayReportsTable = (props: DayReportsTableProps) => {
    * Gets all day reports
    */
   const loadDayReports = async () => {
-    const data = await DayReportDAL.getDayReports(props.way.uuid);
-    setDayReports(data);
+    setDayReports(way.dayReports);
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/reportsTable/DayReportsTable.tsx:48]: don't see any load-related logic

File: mw-webapp/src/service/DayReportService.ts [hunk 2]
```diff
@@ -87,6 +70,8 @@ export class DayReportService {
 
     const validatedDayReportsDTO = DayReportsDTOSchema.parse(dayReportsDTO);
 
+    logRequest({data: validatedDayReportsDTO, text: "ValidatedDayReportsDTO", requestOperation: RequestOperations.READ});
```
Comment (tritonJS826) [inline mw-webapp/src/service/DayReportService.ts:73]: too strange interface

File: mw-webapp/src/utils/logRequest.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
+/* eslint-disable no-console */
+
+/**
+ * Request operations
+ */
+export enum RequestOperations {
```
Comment (tritonJS826) [inline mw-webapp/src/utils/logRequest.ts:6]: why does it logger related?

File: mw-webapp/src/logic/wayPage/reportsTable/DayReportsTable.tsx (hunk 1 above)
Comment (Ekaterina1994) [reply mw-webapp/src/logic/wayPage/reportsTable/DayReportsTable.tsx]: Deleted function

File: mw-webapp/src/service/DayReportService.ts (hunk 2 above)
Comment (Ekaterina1994) [reply mw-webapp/src/service/DayReportService.ts]: Refactored function

File: mw-webapp/src/utils/logRequest.ts (hunk 3 above)
Comment (Ekaterina1994) [reply mw-webapp/src/utils/logRequest.ts]: Deleted

File: mw-webapp/src/logic/wayPage/reportsTable/DayReportsTable.tsx [hunk 4]
```diff
@@ -59,7 +51,7 @@ export const DayReportsTable = (props: DayReportsTableProps) => {
   };
 
   useEffect(() => {
-    loadDayReports();
+    setDayReports(way.dayReports);
     loadMentors();
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/reportsTable/DayReportsTable.tsx:55]: don't load mentors hear (nested components)

File: mw-webapp/src/service/DayReportService.ts [hunk 5]
```diff
@@ -87,6 +70,8 @@ export class DayReportService {
 
     const validatedDayReportsDTO = DayReportsDTOSchema.parse(dayReportsDTO);
 
+    logToConsole(`GetDayReportsDTO required ${validatedDayReportsDTO.length} READ operations`);
```
Comment (tritonJS826) [inline mw-webapp/src/service/DayReportService.ts:73]:
```suggestion
    logToConsole(`DayReportService:${methodName}: ${validatedDayReportsDTO.length} operations`);
```

File: mw-webapp/src/utils/logToConsole.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+/* eslint-disable no-console */
+
+/**
+ * Return in console {@link textToLog}
+ */
+export const logToConsole = (textToLog: string) => {
+  if (process.env.IS_LOGGER_ENABLED === "true") {
+    console.log(`${textToLog}`);
```
Comment (tritonJS826) [inline mw-webapp/src/utils/logToConsole.ts:8]:
```suggestion
    console.log(`${`${textToLog}`}`);
```

Comment (tritonJS826) [reply mw-webapp/src/utils/logToConsole.ts:8]: BTW: it is pretty useful whaen logger could log time (it could help to troubleshoot somitemes)

Comment (Ekaterina1994) [reply mw-webapp/src/utils/logToConsole.ts]: Adde time to log

File: mw-webapp/src/service/DayReportService.ts (hunk 5 above)
Comment (Ekaterina1994) [reply mw-webapp/src/service/DayReportService.ts]: Done

File: mw-webapp/src/logic/wayPage/reportsTable/DayReportsTable.tsx [hunk 7]
```diff
@@ -42,25 +42,17 @@ export const DayReportsTable = (props: DayReportsTableProps) => {
   const isPossibleCreateDayReport = isOwner && isReportForTodayIsNotCreated;
 
   /**
-   * Gets all day reports
+   * Create hashmap with mentors and set value to state
```
[the remark this answers is not in the document]
Comment (Ekaterina1994) [reply mw-webapp/src/logic/wayPage/reportsTable/DayReportsTable.tsx]: Deleted function in DayReport

File: mw-webapp/src/logic/wayPage/reportsTable/DayReportsTable.tsx [hunk 8]
```diff
@@ -42,25 +42,17 @@ export const DayReportsTable = (props: DayReportsTableProps) => {
...
-  };
-
-  /**
-   * Load mentors
-   */
-  const loadMentors = () => {
+  const createHashMapWithMentors = () => {
     const mentorsList = props.way.mentors;
     const mentorsHashMap = new Map(mentorsList.map((item): [string, UserPreview] => [item.uuid, item]));
     setMentors(mentorsHashMap);
```
[the remark this answers is not in the document]
Comment (Ekaterina1994) [reply mw-webapp/src/logic/wayPage/reportsTable/DayReportsTable.tsx]: Updated Way model . Converter array of mentors to hashMap

File: mw-webapp/src/dataAccessLogic/DTOToBusinessConverter/wayDTOToWayPreviewConverter.ts [hunk 9]
```diff
@@ -17,7 +17,7 @@ interface WayProps {
   /**
    * Way's current mentors
    */
-  mentors: UserPreview[];
+  mentors: Map<string, UserPreview>;
```
Comment (tritonJS826) [inline mw-webapp/src/dataAccessLogic/DTOToBusinessConverter/wayDTOToWayPreviewConverter.ts:20]: is it any string?

File: mw-webapp/src/dataAccessLogic/WayDAL.ts [hunk 10]
```diff
@@ -62,15 +62,19 @@ export class WayDAL {
       favoriteForUsersPromise,
     ]);
 
+    const mentorsDictionary = new Map(mentors.map((item): [string, UserPreview] => [item.uuid, item]));
```
Comment (tritonJS826) [inline mw-webapp/src/dataAccessLogic/WayDAL.ts:65]: generic

File: mw-webapp/src/logic/wayPage/MentorRequestsSection.tsx [hunk 11]
```diff
@@ -20,7 +20,7 @@ const addMentorToWay = (
 
   UserPreviewDAL.updateUserPreview(newUserPreview);
 
-  const mentors = way.mentors.concat(newUserPreview);
+  const mentors = way.mentors.set(newUserPreview.uuid, newUserPreview);
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/MentorRequestsSection.tsx:23]: check

File: mw-webapp/src/logic/wayPage/WayPage.tsx [hunk 12]
```diff
@@ -212,7 +212,7 @@ export const WayPage = (props: WayPageProps) => {
   const isWayInFavorites = user && user.favoriteWays.includes(way.uuid);
 
   const isOwner = !!user && user.uuid === way.owner.uuid;
-  const isMentor = !!user && way.mentors.some((mentor) => mentor.uuid === user.uuid);
+  const isMentor = !!user && Array.from(way.mentors.keys()).some((mentorUuid) => mentorUuid === user.uuid);
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/WayPage.tsx:215]: reafactor
wrong type

File: mw-webapp/src/logic/wayPage/reportsTable/DayReportsTable.tsx [hunk 13]
```diff
@@ -41,26 +41,9 @@ export const DayReportsTable = (props: DayReportsTableProps) => {
...
-    const mentorsList = props.way.mentors;
-    const mentorsHashMap = new Map(mentorsList.map((item): [string, UserPreview] => [item.uuid, item]));
-    setMentors(mentorsHashMap);
-  };
-
   useEffect(() => {
-    loadDayReports();
-    loadMentors();
+    setDayReports(way.dayReports);
+    setMentors(way.mentors);
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/reportsTable/DayReportsTable.tsx:46]: why do we need this state here?

File: mw-webapp/src/utils/RequestOperations.ts [hunk 14]
```diff
@@ -0,0 +1,20 @@
+/**
+ * Request operations
+ */
+export enum RequestOperations {
```
Comment (tritonJS826) [inline mw-webapp/src/utils/RequestOperations.ts:4]: it is not util

File: mw-webapp/src/utils/logToConsole.ts [hunk 15]
```diff
@@ -0,0 +1,13 @@
+/* eslint-disable no-console */
+
+/**
+ * Return in console {@link textToLog}
+ */
+export const logToConsole = (textToLog: string) => {
+  const time = new Date().toISOString();
+  if (process.env.IS_LOGGER_ENABLED === "true") {
+    console.log(`${time}: ${textToLog}`);
```
Comment (tritonJS826) [inline mw-webapp/src/utils/logToConsole.ts:9]: no need spaces in meta

File: mw-webapp/src/utils/logToConsole.ts [hunk 16]
```diff
@@ -0,0 +1,13 @@
...
+  } else {
+    return;
+  }
```
Comment (tritonJS826) [inline mw-webapp/src/utils/logToConsole.ts:12]: do we need to return undefined?

**Review by tritonJS826 — APPROVED** [review]

File: mw-webapp/src/dataAccessLogic/DTOToBusinessConverter/wayDTOToWayPreviewConverter.ts (hunk 9 above)
Comment (Ekaterina1994) [reply mw-webapp/src/dataAccessLogic/DTOToBusinessConverter/wayDTOToWayPreviewConverter.ts:22]: Added comment

File: mw-webapp/src/dataAccessLogic/WayDAL.ts (hunk 10 above)
Comment (Ekaterina1994) [reply mw-webapp/src/dataAccessLogic/WayDAL.ts]: Created generic

File: mw-webapp/src/logic/wayPage/MentorRequestsSection.tsx (hunk 11 above)
Comment (Ekaterina1994) [reply mw-webapp/src/logic/wayPage/MentorRequestsSection.tsx:23]: Checked

File: mw-webapp/src/logic/wayPage/WayPage.tsx (hunk 12 above)
Comment (Ekaterina1994) [reply mw-webapp/src/logic/wayPage/WayPage.tsx]: Used .has method

File: mw-webapp/src/logic/wayPage/reportsTable/DayReportsTable.tsx (hunk 13 above)
Comment (Ekaterina1994) [reply mw-webapp/src/logic/wayPage/reportsTable/DayReportsTable.tsx]: Deleted mentors state

File: mw-webapp/src/utils/RequestOperations.ts (hunk 14 above)
Comment (Ekaterina1994) [reply mw-webapp/src/utils/RequestOperations.ts]: Moved enum ro Services

File: mw-webapp/src/utils/logToConsole.ts (hunk 15 above)
Comment (Ekaterina1994) [reply mw-webapp/src/utils/logToConsole.ts]: Deleted

File: mw-webapp/src/utils/logToConsole.ts (hunk 16 above)
Comment (Ekaterina1994) [reply mw-webapp/src/utils/logToConsole.ts:10]: Deleted

**Review by SergioAJS — APPROVED** [review]

**Pull request merged**